src/autogpt_plugins/autogpt_gradio_tools/__init__.py

```diff
@@ -0,0 +1,240 @@
+"""Integration with Gradio Spaces On HuggingFace via gradio_tools."""
+from typing import Any, Dict, List, Optional, Tuple, TypedDict, TypeVar
+from dotenv import load_dotenv
+from auto_gpt_plugin_template import AutoGPTPluginTemplate
+from pathlib import Path
+import os
+from .tools import (AutoGPTClipInterrogatorTool,
+                    AutoGPTStableDiffusion,
+                    AutoGPTWhisperTool,
+                    AutoGPTTextToVideoTool,
+                    AutoGPTCaptioner,
+                    AutoGPTPromptGeneratorTool,
+                    AutoGPTImageToMusicTool,
+                    AutoGPTDocumentAnsweringTool)
+from gradio_tools import GradioTool
+
+PromptGenerator = TypeVar("PromptGenerator")
+
+if (Path(os.getcwd()) / ".env").exists():
+    with open(str(Path(os.getcwd()) / ".env"), 'r') as fp:
+        load_dotenv(stream=fp)
+
+
+TOOLS = [
+    AutoGPTStableDiffusion(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTCaptioner(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTWhisperTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTTextToVideoTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTPromptGeneratorTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTDocumentAnsweringTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTImageToMusicTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTClipInterrogatorTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN"))
+]
+
+def get_tool(tool: str) -> GradioTool:
+    return next(t for t in TOOLS if t.name == tool)
+
+
+class Message(TypedDict):
+    role: str
+    content: str
+
+
+class AutoGPTGradioTools(AutoGPTPluginTemplate):
+    """
+    Twitter API integrations using Tweepy
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._name = "autogpt-gradio-tools"
+        self._version = "0.1.0"
+        self._description = "Calling Gradio Apps ."
+        ts = [t for t in os.getenv("AUTOGPT_GRADIO_TOOLS", "").split(",") if t != ""]
+        self.tools = [get_tool(t) for t in ts]
```

Comment:
Filter attempts to load this to names matching a tool. If you attempt to load an invalid name it will crash

```
@ntindle ➜ /workspaces/Auto-GPT (master) $ AUTOGPT_GRADIO_TOOLS=StableDiffusion,AutoGPTCaptioner ./run.sh
All packages are installed.
NEWS:  Welcome to Auto-GPT! We'll keep you informed of the latest news and features by printing messages here. If you don't wish to see this message, you can run Auto-GPT with the --skip-news flag # INCLUDED COMMAND 'send_tweet' IS DEPRICATED, AND WILL BE REMOVED IN THE NEXT STABLE RELEASE Base Twitter functionality (and more) is now covered by plugins: https://github.com/Significant-Gravitas/Auto-GPT-Plugins ## Changes to Docker configuration The workdir has been changed from /home/appuser to /app. Be sure to update any volume mounts accordingly.
WARNING:  You are running on `master` branch - this is not a supported branch.
Loaded as API: https://gradio-client-demos-stable-diffusion.hf.space ✔
Loaded as API: https://taesiri-blip-2.hf.space ✔
Loaded as API: https://abidlabs-whisper.hf.space ✔
Loaded as API: https://damo-vilab-modelscope-text-to-video-synthesis.hf.space ✔
Loaded as API: https://microsoft-promptist.hf.space ✔
Loaded as API: https://abidlabs-docquery.hf.space ✔
Loaded as API: https://fffiloni-img-to-music.hf.space ✔
Loaded as API: https://pharma-clip-interrogator.hf.space ✔
WARNING: Plugin AutoGPTGradioTools found. But not in the allowlist... Load? (y/n): y
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/workspaces/Auto-GPT/autogpt/__main__.py", line 5, in <module>
    autogpt.cli.main()
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1635, in invoke
    rv = super().invoke(ctx)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/workspaces/Auto-GPT/autogpt/cli.py", line 90, in main
    run_auto_gpt(
  File "/workspaces/Auto-GPT/autogpt/main.py", line 104, in run_auto_gpt
    cfg.set_plugins(scan_plugins(cfg, cfg.debug_mode))
  File "/workspaces/Auto-GPT/autogpt/plugins.py", line 229, in scan_plugins
    loaded_plugins.append(a_module())
  File "/home/vscode/.local/lib/python3.10/site-packages/abstract_singleton/__init__.py", line 14, in __call__
    cls._instances[cls] = super().__call__(*args, **kwargs)
  File "plugins/Auto-GPT-Plugins.zip/Auto-GPT-Plugins-add-gradio-tools/src/autogpt_plugins/autogpt_gradio_tools/__init__.py", line 55, in __init__
  File "plugins/Auto-GPT-Plugins.zip/Auto-GPT-Plugins-add-gradio-tools/src/autogpt_plugins/autogpt_gradio_tools/__init__.py", line 55, in <listcomp>
  File "plugins/Auto-GPT-Plugins.zip/Auto-GPT-Plugins-add-gradio-tools/src/autogpt_plugins/autogpt_gradio_tools/__init__.py", line 36, in get_tool
StopIteration
Press any key to continue...
```